subprojects/docs/src/docs/userguide/workingWithDependencies.adoc

```diff
@@ -15,73 +15,47 @@
-        </sample>
+<sample id="iterating-dependencies" dir="userguide/dependencies/iteratingDependencies" title="Iterating over the dependencies assigned to a configuration">
+    <sourcefile file="build.gradle" snippet="iteration-task" />
+</sample>
 ++++
 
-The `dependencies` task returns only the dependencies belonging explicitly to the configuration. The `allDependencies` task includes the dependencies from extended configurations.
-
-To get the library files of the configuration dependencies you can do:
+=== Iterating over artifacts resolved for a module
 
-++++
-<sample id="configurationHandlingAllFiles" dir="userguide/artifacts/configurationHandling" title="Configuration.files">
-            <sourcefile file="build.gradle" snippet="allFiles"/>
-            <output args="-q allFiles"/>
-        </sample>
-++++
+None of the <<inspecting_dependencies,dependency reporting>> helps you with inspecting or further processing the underlying, resolved artifacts of a module. A typical use case for wanting to access the artifacts is to copy them into a specific directory or filter out files of interest based on a specific file extension.
 
-Sometimes you want the library files of a subset of the configuration dependencies (e.g. of a single dependency).
+You can iterate over the complete set of artifacts resolved for a module with the help of the method api:org.gradle.api.file.FileCollection#getFiles()[]. Using this method on a `Configuration` instance is possible as the interface extends `FileCollection`.
 
 ++++
-<sample id="configurationHandlingFiles" dir="userguide/artifacts/configurationHandling" title="Configuration.files with spec">
-            <sourcefile file="build.gradle" snippet="files"/>
-            <output args="-q files"/>
-        </sample>
+<sample id="iterating-artifacts" dir="userguide/dependencies/iteratingArtifacts" title="Iterating over the artifacts resolved for a module">
+    <sourcefile file="build.gradle" snippet="iteration-task" />
+</sample>
 ++++
 
-The `Configuration.files` method always retrieves all artifacts of the _whole_ configuration. It then filters the retrieved files by specified dependencies. As you can see in the example, transitive dependencies are included.
+[NOTE]
+====
+Iterating over the artifacts of a module automatically resolves the configuration. A resolved configuration becomes immutable and cannot add or remove dependencies. If needed you can copy a configuration for further modification via api:org.gradle.api.artifacts.Configuration#copy()[].
+====
 
-You can also copy a configuration. You can optionally specify that only a subset of dependencies from the original configuration should be copied. The copying methods come in two flavors. The `copy` method copies only the dependencies belonging explicitly to the configuration. The `copyRecursive` method copies all the dependencies, including the dependencies from extended configurations.
+=== Accessing a module's metadata artifact
```

Comment:
"Accessing a module’s metadata artifact" -> "Accessing a module’s metadata file"

We should not talk about _metada artifacts_ as an artifact is something specific for us. Talk about _metadata_ or _metadata files_. I know that we have `MavenPomArtifact` in the API. Still I would avoid the term in the text.

Reply:
I see, ok. done